client/src/util/locationConverter.ts

```diff
@@ -0,0 +1,47 @@
+import { COORDINATE_PRECISION } from '../components/SearchBar/constants.ts';
+import { Location } from '../gql/graphql.ts';
+
+const DOUBLE_PATTERN = '-{0,1}\\d+(\\.\\d+){0,1}';
+
+const LAT_LON_PATTERN = '(' + DOUBLE_PATTERN + ')(\\s*,\\s*|\\s+)(' + DOUBLE_PATTERN + ')';
+
+const ID_SEPARATOR = ':';
```

Comment:
Sorry it took me a while to review this, but in the Transmodel API the IDs are not required to have a colon in it (although in practise they very often do).

Do we need the check for the valid feed id? We could simply say that if it's not a coordinate, then it's quay id, couldn't we?

Reply:
I removed the ID checking part, so any input, that not a valid coordinate pair is processed as an ID input.

On the other hand, I needed to implement error handling around the coordinate and trip plan query, because if the id was invalid, the queries threw an error. The simplest fix was a try-catch around them and an error console log (until a more precise error handling is implemented).